### DISCUSSION TITLE
Many to many join tables?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [aarondl/sqlboiler#985] [Q&A] [answered]
In the Getting Started (v3) video there is an example of using a join table like so:

```sql
create table tags_videos (
  video_id int not null,
  tag_id int not null,
  primary key (video_id, tag_id),
  foreign key (video_id) references videos (id),
  foreign key (tag_id) references tags (id)
);
```

Now obviously this is good for finding all of the tags a video has, but what if we want to find all of the videos that a tag is associated with?

Naturally, I would want to create a secondary index: `CREATE INDEX ON tags_videos(tag_id);`

My question is would sqlboiler transparently pick up on this direction of the join table as well? I know it will for the videos -> tags as shown in the example.

**Comment by aarondl** [comment, marked as answer]
sqlboiler does both directions when it sees something like this. Also the index is optional and varies by use case, you don't need to create one for sqlboiler to understand that, the join table itself is enough.

**Reply by danthegoodman1** [reply]
So just to clarify: in the absence of an index it will just scan the table?

**Reply by aarondl** [reply]
I'm unsure if the partial index could be used instead of having to add a second one now that I think about it. Regardless that index isn't necessary for sqlboiler to understand there's a join table and generate all the proper code to use it. Whether or not you could add that index to increase lookup performance is another issue :D
